public/controllers/dev-tools/dev-tools.js

```diff
@@ -378,43 +378,152 @@ export class DevToolsController {
+            }
+            return `"${parameter.name}": ${valueBodyParam}`;
+          };
+
+          const getInnerKeysBodyRequest = () => {
+            let jsonBodyKeyCurrent = [];
+            let jsonBodyKeyCurrentPosition = {
+              start : {line: currentGroup.start, ch: 0},
+              end: {line: currentGroup.start, ch: 0}
+            };
+            return ([...Array(currentGroup.end + 1 - currentGroup.start).keys()])
+              .reduce((jsonBodyKeyCursor, lineNumberRange) => {
+                const editorLineNumber = currentGroup.start + lineNumberRange;
+                const editorLineContent = editor.getLine(editorLineNumber);
+                const openBracket = editorLineContent.indexOf('{');
+                const closeBracket = editorLineContent.indexOf('}');
+                const keyOpenBracket = (editorLineContent.match(/\s*"(\S+)"\s*:\s*\{/) || [])[1];
+                keyOpenBracket && jsonBodyKeyCurrent.push(keyOpenBracket) && (jsonBodyKeyCurrentPosition.start = {line: editorLineNumber, ch: openBracket});
+
+                closeBracket !== -1 && (jsonBodyKeyCurrentPosition.end = {line: editorLineNumber, ch: closeBracket});
+                if(!jsonBodyKeyCursor && editorCursor.line > jsonBodyKeyCurrentPosition.start.line
+                  && editorCursor.line < jsonBodyKeyCurrentPosition.end.line){
+                    jsonBodyKeyCursor = [...jsonBodyKeyCurrent];
+                };
+                closeBracket !== -1 && jsonBodyKeyCurrent.pop();
+                return jsonBodyKeyCursor
+              }, false);
           }
+          const getInnerPropertyBodyParamObject = (object, keys) => {
+            if(!keys || !keys.length){
+              return object;
+            }
+            const key = keys.shift();
+            if(!object.properties || !object.properties[key] || object.properties[key].type !== 'object'){
+              return [];
+            }
+            return getInnerPropertyBodyParamObject(object.properties[key], keys);
+          };
+
+          if(apiEndpoint && apiEndpoint.body && reLineStart.test(line)){
+            let inputBodyPreviousKeys;
+            let paramsBody = apiEndpoint.body;
+            let requestBodyCursorKeys;
+            if(apiEndpoint.body[0].type === 'object'){
+              requestBodyCursorKeys = getInnerKeysBodyRequest();
+              const paramInnerBody = getInnerPropertyBodyParamObject(apiEndpoint.body[0], [...requestBodyCursorKeys]);
+              paramsBody = Object.keys(paramInnerBody.properties).sort().map(keyBodyParam => ({name:keyBodyParam, ...paramInnerBody.properties[keyBodyParam]}));
+            };
+            try{
+              const bodySanitizedBodyParam = currentGroup.requestTextJson.replace(/(,\s*"\S*\s*)\}/g, '}')/*.replace(/\{(\s*"\S*\s*)(?:["\}]?)/g, '{')*/;
```

Comment:
I will remove this. It is a testing regex to parse the properties of the body when trying to add property before others were previously defined. This would filter the suggestions but it is giving some problems.

For example:
```
GET /cluster/api/config
{
  "access": {
    "| < Cursor is here
    "prop1": "value"
  }
}